### PR TITLE
refactor: use grid density seeding

### PR DESCRIPTION
**Pull request by felix-russo**
Replace gaussian vertex seeding (Sec. 5.4.2 in Bastian Schlag's thesis) with grid vertex seeding (Sec. 5.4.3 in Bastian Schlag's thesis)

### CODE REVIEW

File: Core/include/Acts/Vertexing/GridDensityVertexFinder.hpp [hunk 1]
```diff
@@ -15,6 +15,8 @@
 #include "Acts/Vertexing/Vertex.hpp"
 #include "Acts/Vertexing/VertexingOptions.hpp"
 
+#include <map>
```
Comment (CarloVarni) [inline Core/include/Acts/Vertexing/GridDensityVertexFinder.hpp:18]: why this addition?

File: Examples/Algorithms/Vertexing/include/ActsExamples/Vertexing/IterativeVertexFinderAlgorithm.hpp [hunk 2]
```diff
@@ -22,7 +22,8 @@
 #include "Acts/Vertexing/HelicalTrackLinearizer.hpp"
 #include "Acts/Vertexing/ImpactPointEstimator.hpp"
 #include "Acts/Vertexing/IterativeVertexFinder.hpp"
-#include "Acts/Vertexing/TrackDensityVertexFinder.hpp"
+//#include "Acts/Vertexing/TrackDensityVertexFinder.hpp"
```
Comment (CarloVarni) [inline Examples/Algorithms/Vertexing/include/ActsExamples/Vertexing/IterativeVertexFinderAlgorithm.hpp:25]: please remove commented pieces of code

**Comment by CarloVarni** [comment]
Hi @felix-russo I see there is the WIP label. Is this ready for review, or is it a draft PR?

**Comment by felix-russo** [comment]
> Hi @felix-russo I see there is the WIP label. Is this ready for review, or is it a draft PR?

Hi @CarloVarni, no, this PR is not ready for review yet! I just wanted to know how the different seed finder affects the physics performance... I will look into this further and let you know when I am done!

**Comment by LuisFelipeCoelho** [comment]
> > Hi @felix-russo I see there is the WIP label. Is this ready for review, or is it a draft PR?
> 
> Hi @CarloVarni, no, this PR is not ready for review yet! I just wanted to know how the different seed finder affects the physics performance... I will look into this further and let you know when I am done!

just to mention that you can also run physmon locally to check the performance, if you prefer

**Comment by felix-russo** [comment]
Will investigate this later

**Pull request closed without merging**